packages/core/src/controllers/SendController.ts

```diff
@@ -228,12 +249,195 @@ export const SendController = {
     }
   },
 
+  async generateLink() {
```

Comment:
This is a huge method. Can we break down in smaller ones and maybe introduce an util for this?